tests/modules/test_fusions.py

```diff
@@ -0,0 +1,101 @@
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_BlockTucker(self):
+        fusion = fusions.BlockTucker(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_Mutan(self):
+        fusion = fusions.Mutan(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_Tucker(self):
+        fusion = fusions.Tucker(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_MLB(self):
+        fusion = fusions.MLB(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_MFB(self):
+        fusion = fusions.MFB(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_MFH(self):
+        fusion = fusions.MFH(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_MCB(self):
+        fusion = fusions.MCB(self.input_dims, self.output_dims, mm_dim=100)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_LinearSum(self):
+        fusion = fusions.LinearSum(self.input_dims, self.output_dims, mm_dim=20)
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+    def test_ConcatMLP(self):
+        fusion = fusions.ConcatMLP(self.input_dims, self.output_dims, dimensions=[5,5])
+        out = fusion(self.x)
+        if torch.cuda.is_available():
+            fusion.cuda()
+            out = fusion([self.x[0].cuda(), self.x[1].cuda()])
+        assert torch.Size([2,2]) == out.shape
+
+if __name__ == '__main__':
```

Comment:
Thanks for writing these great tests. Though these tests don't test the working of the fusion layer itself. You should set a seed in start and then compare the output values to pre-saved values for results to make sure something hasn't broken.

Reply:
Do I need to check the output as well by setting manual seed and assert almost equal with the hardcoded values as in the [layer tests](https://github.com/facebookresearch/pythia/blob/2ab8083f3da6a5421c5f5312df3b4ca7cccb230c/tests/modules/test_layers.py#L25)

Reply:
Yes, that should be case. Let's fix that in a separate PR.